### PR TITLE
US013 - Add new Capability

### PR DESCRIPTION
**Pull request by TheoHodges**
As an Admin
I want to be able to create a new capability and assign it existing bands
So that I can keep the system up to date

Acceptance criteria

- Able to add a role into a new capability
- Able to define:
 - New capability name

### CODE REVIEW

**Review by sebastianczajkowski — CHANGES_REQUESTED** [review]
Is not having any validation intended?

**Comment by TheoHodges** [comment]
> Is not having any validation intended?
What validation would you be looking for?

File: tests/capability.test.js [hunk 1]
```diff
@@ -40,4 +42,53 @@ describe("Test getting capabilities", () => {
...
+
+const newCapabilityError = {
+    name: "<script>maliciousCode();</script>",
+};
+
+describe("Test adding a new capability", () => {
+
+    test("The capability creation should be successful", async () => {
+        let expected = { status: 200 }
+
```
Comment (jkelso662) [inline tests/capability.test.js:59]: Lots of white space here. The only white space I would want in this class is between tests, or if you wanted to separate the three steps of a test out within a test function: setup, call, assert.

E.g. 
```
test("something", async () => {
    setup ...
    setup ...

    something.call ...

    assert ...
    assert ...
}
```

**Review by jkelso662 — CHANGES_REQUESTED** [review]

**Review by mactrz — APPROVED** [review]
Looks good

File: services/capabilityService.js [hunk 2]
```diff
@@ -14,4 +10,40 @@ exports.getCapabilityLeads = async () => {
...
+    let capabilities;
+    await axios.get(`http://localhost:8080/capability/`)
+        .then(response => {
+            capabilities = response.data;
+        })
+        .catch(error => {
+            console.log(error);
+        });
+    if (capabilities) {
+        capabilities.forEach(function (capability) {
```
Comment (sebastianczajkowski) [inline services/capabilityService.js:42]: Just a heads-up - current solution is fine but costly, below is what I would do

You could use a method that searches for one specific capability by name and if it doesn't find anything, then let the user create a new one, but if it returns a capability, then show the user an error.

**Review by sebastianczajkowski — APPROVED** [review]
The solution I provided is completely optional, so it's good as it is

File: services/capabilityService.js (hunk 2 above)
Comment (TheoHodges) [reply services/capabilityService.js:42]: I think I will make this change

**Review by jkelso662 — APPROVED** [review]
Much better! 🤠

**Pull request closed without merging**